Add support for GraphicsMagick

cblock opened the pull request:
Attached please find my initial attempt at integrating GraphicsMagick into the lib as proposed in #7.

Please note that there GraphicsMagick handles EXIF data slightly differently. Instead of providing `exif:DateTimeOriginal` keys it provides `DateTimeOriginal` keys. I thus added a little normalization logic that ensures EXIF Data ist always provided in GraphicsMagick format no matter what library is used underneath.

cblock committed:
Add support for GraphicsMagick

cblock committed:
Add default configs + specifig graphicsmagick config and adjust travis ci to run graphicsmagick variant

cblock commented:
Not quite sure why travis fails. Looks like GM executable cannot be called though graphicsmagick seem to have been installed properly. @keichan34 maybe you have an idea?

keichan34 commented on lib/exfile_imagemagick/sys_runner.ex:
```diff
@@ -1,9 +1,20 @@
   alias ExfileImagemagick.Server
 
-  def cmd(command, args) do
+  @image_processor Application.get_env(:exfile_imagemagick, :image_processor, :imagemagick)
```
If you move this to `ExfileImagemagick.Server`, the code could be cleaned up a bit.

keichan34 commented on lib/exfile_imagemagick/sys_runner.ex:
```diff
@@ -1,9 +1,20 @@
-  def cmd(command, args) do
+  @image_processor Application.get_env(:exfile_imagemagick, :image_processor, :imagemagick)
+
+  def generic_cmd(command, args) do
```
If generic_cmd won't be used, we don't need it

keichan34 reviewed (commented):
It looks like there are a couple ways to add graphicsmagick to the Travis build -- can you try a couple of them out?

for example, https://github.com/travis-ci/travis-ci/issues/3128

cblock committed:
Trying different way of adding graphicsmagick to travis-ci

cblock committed:
Try travis-ci setup of https://github.com/aheckmann/gm

cblock committed:
Move image_processor specific logic from sys_runner to GenServer and change image_cmd -> cmd

cblock committed:
Add elixir 1.5.2 / otp 20.1 to the test matrix

cblock committed:
Calling graphicsmagick on linux distors requires lower case "gm" command instead of uppercase "GM"

cblock committed:
Add newer graphicsmgick dependency to travis build env

cblock committed:
Force "yes" on add-apt-repository command in travis setup

cblock commented:
@keichan34 Thanks for your code review and the suggestions. I moved the image procesor logic to the GenServer and I also fixed the failing tests in Travis CI.

keichan34 commented:
@cblock Looks good! Thanks!
